### PR TITLE
[StyleCop] Review and fix warnings on BotkitConversation.cs

### PR DESCRIPTION
**Pull request by Konigstiger**
**Changes made**

-Add documentation for classes and methods.
-Remove multiple blank lines.

Note: ‘async method lacks 'await' operators and will run synchronously’ and ‘The file header XML is invalid’ rules were ignored.

### CODE REVIEW

File: Botkit/Microsoft.BotKit/Conversation/BotkitConversation.cs [hunk 1]
```diff
@@ -1,46 +1,47 @@
...
 // Licensed under the MIT License.
 
-using Microsoft.BotKit.Core;
-using Microsoft.Bot.Builder.Dialogs;
-using System;
-using System.Threading.Tasks;
-
 namespace Microsoft.BotKit.Conversation
 {
+    using System;
```
Comment (ceciliaavila) [inline Botkit/Microsoft.BotKit/Conversation/BotkitConversation.cs]: Please, keep the using statements outside the namespace's scope. This rule should be ignored.

File: Botkit/Microsoft.BotKit/Conversation/BotkitConversation.cs [hunk 2]
```diff
@@ -1,46 +1,47 @@
...
+    using System.Threading.Tasks;
+    using Microsoft.Bot.Builder.Dialogs;
+    using Microsoft.BotKit.Core;
```
Comment (Maihav) [inline Botkit/Microsoft.BotKit/Conversation/BotkitConversation.cs]: We are ignoring the rule regarding the _using_ inside a _namespace_. Can you revert this change?

File: Botkit/Microsoft.BotKit/Conversation/BotkitConversation.cs [hunk 3]
```diff
@@ -129,81 +130,79 @@ public void Before(string threadName, Object obj)
...
         {
-
         }
 
         /// <summary>
         /// This private method is called at the end of the conversation, and causes any bound handler functions to be executed.
         /// </summary>
-        /// <param name="context">The current dialog context</param>
-        /// <param name="any">An object containing the final results of the dialog</param>
-        public async void RunAfter(DialogContext context, object any)
```
Comment (ceciliaavila) [inline Botkit/Microsoft.BotKit/Conversation/BotkitConversation.cs]: Please keep the **_async_** indicator. When this method gets implemented, it will need it.

File: Botkit/Microsoft.Bot.Sample.Slack/Startup.cs [hunk 4]
```diff
@@ -36,7 +35,8 @@ public void ConfigureServices(IServiceCollection services)
 
             // Create the options for the SlackAdapter
             services.AddSingleton<ISlackAdapterOptions, ConfigurationSlackAdapterOptions>();
-            //services.AddSingleton<IBotFrameworkHttpAdapter, BotFrameworkHttpAdapter>();
+
+            // services.AddSingleton<IBotFrameworkHttpAdapter, BotFrameworkHttpAdapter>();
```
Comment (GasparAcevedoZainSouthworks) [inline Botkit/Microsoft.Bot.Sample.Slack/Startup.cs]: Can you remove this line, as this is only commented code?

**Comment by VictorGrycuk** [comment]
Just a little feedback regarding the formatting of the PR's details.
- Add a space after the `-` to create bullets points
- I would make `Note:` in bold text, **Note:**, so it stands out more

The rest looks good to me 👍

**Comment by GasparAcevedoZainSouthworks** [comment]
I Just noticed that there are conflicts that must be resolved. Please, take this one first

File: libraries/Microsoft.BotKit/Conversation/BotkitConversation.cs [hunk 5]
```diff
@@ -129,81 +130,79 @@ public void Before(string threadName, Object obj)
...
         {
-
         }
 
         /// <summary>
         /// This private method is called at the end of the conversation, and causes any bound handler functions to be executed.
         /// </summary>
-        /// <param name="context">The current dialog context</param>
-        /// <param name="any">An object containing the final results of the dialog</param>
-        public async void RunAfter(DialogContext context, object any)
```
Comment (ceciliaavila) [inline libraries/Microsoft.BotKit/Conversation/BotkitConversation.cs]: Let's keep the _**async**_ here.

File: Botkit/Microsoft.BotKit/Conversation/BotkitConversation.cs (hunk 3 above)
Comment (Maihav) [reply Botkit/Microsoft.BotKit/Conversation/BotkitConversation.cs]: Done

File: libraries/Microsoft.BotKit/Conversation/BotkitConversation.cs (hunk 5 above)
Comment (Maihav) [reply libraries/Microsoft.BotKit/Conversation/BotkitConversation.cs]: Done
